labs/content/labs/06-capstone.md

```diff
@@ -0,0 +1,427 @@
+---
+title: Capstone
+index: 6
+---
+
+# Lab Six - Capstone
+
+## `cd` to the sixth lab
+
+* In a terminal:
+
+
+```bash
+cd ../ # presuming still in first lab
+cd lab-06-capstone
+yarn start
+```
+
+### Are we done yet?
+
+The application is pretty much feature-complete at this point. Nice work!
+
+...but then you notice that just about anybody can modify data. Not good.
+We need to add some security to this app, fast!
+
+### What do we need?
+
+We need to restrict access by requiring the user log in using valid credentials.
+The good news is that our server team has put together a nice login/logout capability.
+
+**Note**: This is a very naïve login/logout capability as a simple example. Don't use it in production code!
+
+### How are we going to do it?
+
+There aren't any new concepts in this section - we're just applying all the things you've learned from previous labs.
+**Try and implement each step yourself** - don't be afraid to ask questions or collaborate with classmates to work out a solution.
+If you truly get stuck then each section has a code hint you can refer to.
+
+### Build the Action Types for Auth
+
+Our app is going to have to accept credentials, log the user in, and pull the current user from the server
+so we can get info about them.
+
+We need to define our types for Authentication actions.
+
+* One type for when we get a copy of the user from the server and need to save it into State
+* A second type to track whether we encountered an error when logging in so we can tell the user.
+
+
+<details>
+  <summary>Code hint:</summary>
+
+```javascript:title=AuthActionTypes.js
+export const SET_USER = 'SET_USER';
+export const ERROR = 'ERROR';
+```
+
+</details>
+
+&nbsp;
+
+
+### Build the Actions
+
+Now we need a way to fire off login & logout actions based on user interaction.
+
+We need async actions for making API calls to the following endpoints:
+* POST `/api/login`, request body of { username: value, password: value }
+* POST `/api/logout`, no request body
+
+Then we need actions:
+
+* To handle getting a copy of the user after login & clearing the user after logout
+* To track errors on login
+
+
+<details>
+  <summary>Code hint:</summary>
+
+```javascript:title=AuthActionCreator.js
```

Comment:
Looks like a mixture of `function` and `=>`. Do we want to switch them all to arrow functions?

Reply:
I was planning to look into doing an overhaul of arrow functions and removing use of `bindActionCreators` once this got merged. More complicated since I have to update the timesheet repo at the same time as the slides